src/CoreBundle/private/modules/CMSTemplateEngine/CMSTemplateEngine.class.php

```diff
@@ -449,22 +451,34 @@ protected function _GetModuleInstanceList()
     }
 
     /**
-     * loads the navi into $data and returns the active menuItem.
-     *
-     * @return string
+     * Filters the main navigation items.
      */
-    protected function GetMainNavigation()
+    protected function filterMainNavigation(): void
     {
         $this->data['oMenuItems'] = $this->oTableManager->oTableEditor->GetMenuItems();
         $this->data['oMenuItems']->RemoveItem('sItemKey', 'save');
         $this->data['oMenuItems']->RemoveItem('sItemKey', 'copy');
         $this->data['oMenuItems']->RemoveItem('sItemKey', 'new');
         $this->data['oMenuItems']->RemoveItem('sItemKey', 'delete');
-        $this->data['oMenuItems']->RemoveItem('sItemKey', 'revisionManagement');
-        $this->data['oMenuItems']->RemoveItem('sItemKey', 'revisionManagementLoad');
         $this->data['oMenuItems']->RemoveItem('sItemKey', 'copy_translation');
-        $this->data['oMenuItems']->RemoveItem('sItemKey', 'edittableconf');
+    }
+
+    /**
+     * @deprecated since 6.3.0 - renamed and splitted to 2 methods: filterMainNavigation and getActiveModuleLayout.
+     */
+    protected function GetMainNavigation()
+    {
+        return '';
```

Comment:
Can we make it so that this method still works as before? Deprecation should mean that a method is still usable until it is removed.

Reply:
reverted the deletion...

Reply:
The two new methods do exactly the same in combination as the old method did before, don't they? Calling those from the old method would remove duplicate code and still preserve functionality.

Reply:
yep... you are right. Done